kuma/users/tests/test_templates.py

```diff
@@ -1296,30 +1255,98 @@ def test_delete_link_does_not_appear_summary_page_other_revisions(self):
+            force_locale=True)
         # TODO: PhaseIV
         # delete_url_new_action
 
-        # TODO: PhaseIII
-        # delete_link_reverted_section = page.find('#reverted a[href="{url}"]'.format(
-        #     url=delete_url_reverted))
+        delete_link_reverted_section = page.find('#reverted a[href="{url}"]'.format(
+            url=delete_url_reverted))
         # TODO: PhaseIV
         # delete_link_new_action_section = page.find('#new-actions-by-user a[href="{url}"]'.format(
         #     url=delete_url_new_action))
         delete_link_already_spam_section = page.find('#already-spam a[href="{url}"]'.format(
             url=delete_url_already_spam))
 
         # There should not be a delete link in any of these sections
-        # TODO: PhaseIII
-        # eq_(len(delete_link_reverted_section), 0)
+        eq_(len(delete_link_reverted_section), 0)
         # TODO: PhaseIV
         # eq_(len(delete_link_new_action_section), 0)
         eq_(len(delete_link_already_spam_section), 0)
 
+    def test_newest_revision_is_not_spam(self):
+        """
+        Test with a spam user who has made revisions to a single document,
+        but another user has made a more recent revision.
+        The newest revision was created by a non-spam user,
+        so none of the revisions actually need to be reverted.
+        """
+        # Create 3 revisions for self.testuser, titled 'Revision 1', 'Revision 2'...
+        spam_revisions = self.create_revisions(
+            num=3,
+            document=self.document,
+            creator=self.testuser)
+        self.create_revisions(
+            num=1,
+            document=self.document,
+            creator=self.admin)
+
+        self.client.login(username='admin', password='testpass')
+        ban_url = reverse('users.ban_user_and_cleanup_summary',
+                          kwargs={'user_id': self.testuser.id})
+        full_ban_url = self.client.get(ban_url)['Location']
+
+        data = {'revision-id': [rev.id for rev in spam_revisions]}
+        resp = self.client.post(full_ban_url, data=data)
+        eq_(200, resp.status_code)
+        page = pq(resp.content)
+
+        revisions_needing_follow_up = page.find('#manual-revert-needed li')
+        revisions_reported_as_spam = page.find('#revisions-reported-as-spam li')
+
+        # Because no revisions needed to be reverted, none need follow up
+        eq_(len(revisions_needing_follow_up), 0)
+        # Only one document is listed on the reported as spam list (distinct document)
+        eq_(len(revisions_reported_as_spam), 1)
+
+    def test_multiple_revisions_are_spam(self):
+        """
+        Test with a spam user who has made multiple revisions to a single document.
+        This document should be reverted to the last version that was created
+        by a non-spam user (self.admin).
+        The newest revision was created by a non-spam user,
+        so none of the revisions actually need to be reverted.
+        """
```

Comment:
I don't think this comment is correct for the test ``test_multiple_revisions_are_spam``. The admin created the first revision, and the next three by the spammer, and then no more.  I think the comment needs to change, and an additional check is needed that there _were_ reversions on this page, like:

```
revisions_reverted = page.find('#revisions-reverted li')
eq_(len(revisions_reverted), 1) 
```